Source/WebCore/platform/graphics/displaylists/DisplayListRecorder.cpp

```diff
@@ -479,13 +479,17 @@ void Recorder::drawFocusRing(const Vector<FloatRect>& rects, float outlineOffset
     recordDrawFocusRingRects(rects, outlineOffset, outlineWidth, color);
 }
 
-void Recorder::fillRect(const FloatRect& rect)
+void Recorder::fillRect(const FloatRect& rect, bool)
 {
+    auto& state = currentState();
+    auto& graphicsState = state.state;
+    bool performRectIntersection = graphicsState.fillBrush().gradient() && !graphicsState.dropShadow();
```

Comment:
Did you notice bugs when drop shadows are present? Did tests detect them? If not, please add a test.

I wonder if there is other graphics context state we should consult.

Reply:
I did not notice any issues with drop shadows as they would not reach the path with the clipping flag is checked. The flag is not consulted when drop shadows are present (see GraphicsContextCG.cpp:820 - `if (hasDropShadow()) {`), so the goal of the check is to avoid the matrix multiplication when it is not needed.

We could compute this when drop shadows are present, but I don't think it is necessary since we will not use the variable.

Reply:
EWS found a failure :) 
imported/mozilla/svg/opacity-and-gradient-02.svg

I will take a look